pandas/io/html.py

```diff
@@ -435,20 +460,22 @@ def row_is_all_th(row):
             while body_rows and row_is_all_th(body_rows[0]):
                 header_rows.append(body_rows.pop(0))
 
-        header = self._expand_colspan_rowspan(header_rows)
+        header = self._expand_colspan_rowspan(header_rows, header=True)
         body = self._expand_colspan_rowspan(body_rows)
         footer = self._expand_colspan_rowspan(footer_rows)
 
         return header, body, footer
 
-    def _expand_colspan_rowspan(self, rows):
+    def _expand_colspan_rowspan(self, rows, header=False):
```

Comment:
whats the reason we need to use `header` arg here? I see the comment about MultiIndex but is it possible to avoid this in another way and still extract hrefs?

If instead it was `extract_links in {'body', 'all', 'header', 'none'}` would this be possible? instead of currently where True is implicitly `body` and False is implicitly `none`

Reply:
I don't think it's possible to avoid the MultiIndex if we read the headers, but I think it's not a bad idea to have it as optional and left to the dev's choice. I'll implement your suggestion instead.